c7n/resources/vpc.py

```diff
@@ -18,13 +18,53 @@
 from c7n.manager import resources
 from c7n.utils import local_session, type_schema
 
+from c7n.filters import ValueFilter, Filter, FilterRegistry
+from c7n.utils import local_session, type_schema
+
 
 @resources.register('vpc')
 class Vpc(QueryResourceManager):
 
     resource_type = 'aws.ec2.vpc'
 
 
+
+@Vpc.filter_registry.register('subnets')
+class SubnetsOfVpc(ValueFilter):
+
+    schema = type_schema('subnets', rinherit=ValueFilter.schema)
+
+    def __init__(self, *args, **kw):
+        super(SubnetsOfVpc, self).__init__(*args, **kw)
+        self.data['key'] = 'Subnets'
+
+    def get_vpc_subnets(self):
+        # from c7n.resources.vpc import Subnet
+        manager = Subnet(self.manager.ctx, {})
+        return [s for s in manager.resources()]
+
+    def process(self, resources, event=None):
+
+        subnets = self.get_vpc_subnets()
+
+        def _user_vpc_subnets(resource):
+            resource['Subnets'] = [s for s in subnets if s['VpcId'] == resource['VpcId']]
+
+        with self.executor_factory(max_workers=1) as w:
```

Comment:
replace the inner and executor factory with just a loop or comprehension, if there are no api calls being done in the executor then it will always be slower than just straight inlining (at least for thread executors in py).